dulwich/repo.py

```diff
@@ -1532,7 +1532,17 @@ def get_blob_normalizer(self):
         """Return a BlobNormalizer object"""
         # TODO Parse the git attributes files
         git_attributes = {}
-        return BlobNormalizer(self.get_config_stack(), git_attributes)
+        config_stack = self.get_config_stack()
+        try:
+            tree = self.object_store[self.refs[b"HEAD"]].tree
```

Comment:
this was also discussed in the original PR https://github.com/dulwich/dulwich/pull/704#discussion_r275740141

rather than comparing paths to what's in the `HEAD` tree, we could check what's in the index's blobs instead (if the index already has CRLF line endings in the existing blob, then assume we aren't supposed to convert it)

but that doesn't seem too much cleaner of a solution to me